packages/now-build-utils/src/detect-builders.ts

```diff
@@ -107,6 +107,31 @@ async function detectApiBuilders(files: string[]): Promise<Builder[]> {
   return finishedBuilds as Builder[];
 }
 
+// When a package has files that conflict with `/api` routes
+// e.g. Next.js pages/api we'll check it here and return an error.
+async function checkConflictingFiles(
+  files: string[],
+  builders: Builder[]
+): Promise<ErrorResponse | null> {
+  // For Next.js
+  if (builders.some(builder => builder.use.startsWith('@now/next'))) {
+    const hasApiPages = files.some(file => file.startsWith('pages/api/'));
+    const hasApiBuilders = builders.some(builder =>
+      builder.src.startsWith('api/')
+    );
+
+    if (hasApiPages && hasApiBuilders) {
```

Comment:
Can we pass `nowJson.routes` into this function?

Then we could add `const hasRoutes = routes && routes.length > 0`.

```suggestion
    if (hasApiPages && hasApiBuilders && !hasRoutes) {
```